frontend/src/modules/search/withSearch.js

```diff
@@ -0,0 +1,56 @@
+/* @flow */
+
+import * as React from 'react';
+import createMarker from 'react-content-marker';
+
+
+export function highlightSearch(base: string, search: string){
+    // Split search string on spaces except if between non-escaped quotes.
+    const unusable = '☠';
+    const searchWords = search.replace(/\\"/g, unusable).match(/[^\s"]+|"[^"]+"/g);
+
+    if (searchWords) {
+        const reg = new RegExp(unusable, 'g');
+        var i = searchWords.length;
+
+        while(i--) {
+            searchWords[i] = searchWords[i].replace(/^["]|["]$/g, '');
+            searchWords[i] = searchWords[i].replace(reg, '"');
+        }
+
+        // Sort array in decreasing order of string length
+        searchWords.sort(function(a, b) {
+            return b.length - a.length;
+        });
+        
+        const rules = searchWords.map(function(searchWord) {
+            return {
+                rule: new RegExp(searchWord, 'i'),
+                tag: searchWord => <mark className='search'>{ searchWord }</mark>,
+            };
+        });
+
+        const WithSearch = createMarker(rules);
+
+        return <WithSearch>{ base }</WithSearch>;
```

Comment:
You don't want to return a React Component instance here, because that won't be parsable by the next `react-content-marker` parser. Instead you want this function to return an array of strings and elements, basically the result of the `mark` function.

However, I don't think that it is accessible from the module. I'll have to release a new version of the library, will do that shortly.

Reply:
See https://github.com/mozilla/react-content-marker/pull/3